### ISSUE TITLE
Focus out not being called on input fields

### ISSUE THREAD

**Issue by dusanstanojeviccs** [html-next/vertical-collection#157]
Bug: Yes

Version: 1.0.0-beta.9

OS: iOS 11

Browser: Safari

Description:
Table row contains input fields and one field is focused. When the focused input goes outside of the currently shown rows "focus-out" is not triggered.

Repo that demonstrates the issue:
https://github.com/dusanstanojeviccs/ember-vc-issue-no-focus-out

Steps to recreating the issue:
1. Get the repo
2. run "ember s"
3. Go to localhost:4200/demo
4. With console open click into the first input field
5. Scroll the table
6. Click into an input field

Expected:
In console it logs the number of focus outs. (in this case 1)

Actual:
In Chrome Desktop behaves as expected, in Safari on iOS nothing is printed to the console.

This issue makes building something similar to excel with VC impossible.

**Comment by pzuraq** [comment]
Thanks for opening this issue! Focus is pretty tricky, I've opened up a separate issue, #160, for what I think needs to happen behavior-wise for us to get this right. I'll probably have time to take this on sometime after the holidays, also glad to help anyone who wants to give it a shot in the meantime!

**Comment by pzuraq** [comment]
@dusanstanojeviccs We've spent a bit more time looking at this recently and it looks like the main issue here is a difference in behavior between Chrome and other browsers. Like you've pointed out, Chrome appears to work fine - when the current active element is removed from the DOM `blur` events fire like normal, but in Safari and Firefox they do not. 

This points to the exact behavior around this being underspec'd (or Chrome is just wrong here), but for us it presents a challenge because if we trigger a blur event manually when we remove an item, then it will trigger _twice_ in Chrome. We don't want to begin branching behavior for different browser versions, so for the time being we won't be able to support this.

In the long run we do want to implement a more radical approach to focus, like we've outlined in #160. Ideally, there should be no perceivable difference in focus behavior than if we actually rendered the content.

**Comment by runspired** [comment]
If an element is focused we should probably just leave it's item in the DOM

**Comment by pzuraq** [comment]
@runspired yes, that’s what I was referring to as a more radical approach. Unfortunately I’m not going to have time to work on that soon, it’s a pretty massive shift in how we do things currently, but that’s the best path forward here.

**Comment by dusanstanojeviccs** [comment]
@pzuraq @runspired Hey, thanks for following up on this. I agree that keeping the focused item in the DOM is the best solution. Hmm I am really busy but I might be able to take a look if you could kinda walk me through how it works or how it should be changed.

**Comment by pzuraq** [comment]
I should have time sometime next week to do a sync and talk you through the overall structure. That change is going to be pretty significant, which is why we were planning on revisiting it after v1.0.0, but definitely happy to walk you through 😄
